src/psyclone/tests/psyir/nodes/array_mixin_test.py

```diff
@@ -691,6 +693,12 @@ def test_get_effective_shape(fortran_reader):
     with pytest.raises(NotImplementedError) as err:
         _ = routine.children[child_idx].lhs._get_effective_shape()
     assert "include a function call or expression" in str(err.value)
+    # Array access with indices given by another array that is not explicitly
+    # indexed.
+    child_idx += 1
+    shape = routine.children[child_idx].lhs._get_effective_shape()
+    assert len(shape) == 1
+    assert "SIZE(a)" in shape[0].debug_string()
```

Comment:
Could you test if we do the right thing if we don't know the type of the symbols:
`b(scalarval, arrayval) = 1`

And I see this method is not overriden by ArrayMember nor ArrayOfStructuresMixing. So we could also add some tests with these constructs here.

Reply:
We didn't but we do now. Added tests.